### ISSUE TITLE
gemma3_text: attention mask cast to packed uint32 weight dtype breaks padding in quantized checkpoints

### ISSUE THREAD

**Issue by alekseysotnikov** [Blaizzy/mlx-embeddings#72]
## Problem

`Model.__call__` in `mlx_embeddings/models/gemma3_text.py` casts the additive attention mask to `self.model.embed_tokens.weight.dtype`:

```python
extended_attention_mask = extended_attention_mask.astype(
    self.model.embed_tokens.weight.dtype
)
```

In quantized checkpoints (e.g. `mlx-community/embeddinggemma-300m-8bit`) that dtype is **packed uint32**. Casting the `0.0 / -inf` mask to uint32 silently destroys the `-inf` entries, so **pad tokens leak into attention** whenever padded batches are used — which is the documented batch-processing path (`batch_encode_plus(padding=True)` + `attention_mask`).

fp16-weight checkpoints work only by accident because their weight dtype happens to be floating.

## Minimal repro

EmbeddingGemma-300m-8bit, one 11-char sentence:

| setup | cos vs embedded alone |
|---|---|
| batched with itself | 1.00000 |
| batched with 127 copies of a max-length sentence (**current code**) | **0.884** |
| same, mask cast to `mx.float16` instead | **0.999997** |

Over a full real-world transcript (395 sentences, natural-order batching): mean cosine vs zero-padding ground truth is **0.928** (385/395 sentences below 0.99) — nearly every embedding is distorted by its batch neighbors' padding.

## Related

- #43 fixed the related symptom (bidirectional masking for padded batches) but also casts to model dtype, so it would not survive uint32-packed checkpoints either.

PR incoming.
